babel.config.js

```diff
@@ -3,7 +3,25 @@
 //   https://babeljs.io/docs/en/config-files
 
 module.exports = {
-  presets: [
-    'module:metro-react-native-babel-preset',
-  ],
+  // Use different presets depending on ${BABEL_ENV:-${NODE_ENV:-development}}.
+  // https://babeljs.io/docs/en/6.26.3/babelrc#env-option
+  env: {
+    // Most of our project is bog-standard React Native. We therefore use the
+    // RN-provided presets by default.
+    development: {
+      presets: ['module:metro-react-native-babel-preset'],
+    },
+
+    // JavaScript injected into the webview runs on a different JS engine and is
+    // packaged with a different bundling mechanism; it therefore needs very
+    // different transformations. The RN-provided presets, in particular, are
+    // completely unsuitable.
+    //
+    // (We explicitly specify the transformations we do want in our build script
+    // for the webview code, `tools/generate-webview-js`, rather than relying on
+    // this config file.)
+    webview: {
+      /* this space intentionally left blank */
+    },
```

Comment:
Not trivially. The only way to avoid loading `babel.config.js` is by changing `$CWD`, and I'd bet lunch that that will break something else. (Passing `babelrc: false` doesn't disable the `babel.config.js` file; it's just for [file-relative configs](https://babeljs.io/docs/en/config-files#file-relative-configuration).)

There's also no way to override it in lower-level files. Project-level options are merged with all other configuration; while we could use a lower-level  `.babelrc` file to override any options passed _to_ `metro-react-native-babel-preset`, we can't override the fact that it was included in the first place.

Alternatively, we could (e.g.) reorganize the file hierarchy to have React-targeted source in one subdirectory and WebView-targeted source in another. I'm not opposed in principle, but it feels like too much of a commitment for this commit series.

Reply:
... I have no idea how I missed that. Done.